### PR TITLE
Problem: Cosmos SDK and ibc-go are not upgraded to `0.45.7` and `4.0.0-rc3`

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Upgrade versions for both `cosmos-sdk` and `ibc-go`

### CODE REVIEW

File: app/test_helpers.go [hunk 1]
```diff
@@ -330,6 +331,7 @@ func SignCheckDeliver(
 ) (sdk.GasInfo, *sdk.Result, error) {
 
 	tx, err := helpers.GenTx(
+		rand.New(rand.NewSource(time.Now().UnixNano())),
```
Comment (yihuang) [inline app/test_helpers.go:334]: probably need a "nolint" here

File: app/test_helpers.go [hunk 2]
```diff
@@ -330,6 +331,7 @@ func SignCheckDeliver(
 ) (sdk.GasInfo, *sdk.Result, error) {
 
 	tx, err := helpers.GenTx(
```
Comment (tomtau) [inline app/test_helpers.go:333]: wouldn't `time.now()` not make it non-deterministic? or are those generated txs not regenerated in simulations?

Comment (devashishdxt) [reply app/test_helpers.go:333]: We only use this in (nft) unit tests.

**Review by tomtau — APPROVED** [review]

**Comment by devashishdxt** [comment]
Forgot to update `gomod2nix`

**Pull request merged**